snow/consensus/snowman/topological.go

```diff
@@ -508,7 +504,7 @@ func (ts *Topological) vote(ctx context.Context, voteStack []votes) (ids.ID, err
 		pollSuccessful = parentBlock.sb.RecordPoll(vote.votes) || pollSuccessful
 
 		// Only accept when you are finalized and the head.
-		if parentBlock.sb.Finalized() && ts.head == vote.parentID {
+		if parentBlock.sb.NumProcessing() == 1 && ts.head == vote.parentID {
```

Comment:
For this PR I think we should leave the `Snowball` interface unmodified. I'm not really sure if we want to modify the `Snowball` interface at all - I mainly wanted to update the `Snowman` interface because we are currently exposing duplicated information.